chart/templates/hook-upgrade-crds.yaml

```diff
@@ -88,9 +88,14 @@ spec:
       serviceAccountName: {{ template "vso.chart.fullname" . }}-upgrade-crds
       securityContext:
         {{- toYaml .Values.controller.podSecurityContext | nindent 8 }}
+      {{- with .Values.controller.imagePullSecrets }}
+      imagePullSecrets:
+      {{- toYaml . | nindent 8 }}
+      {{- end }}
```

Comment:
How about setting the pull secrets on the service account, like we do for [the deployment's service account](https://github.com/hashicorp/vault-secrets-operator/blob/be6ff476c51d98aa78b3290e62ac3271813df38e/chart/templates/deployment.yaml#L14)?

i.e.

```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: {{ template "vso.chart.fullname" . }}-upgrade-crds
  namespace: {{ .Release.Namespace }}
  labels:
    app.kubernetes.io/component: controller-manager
{{ include "vso.chart.labels" . | indent 4 }}
  annotations:
    helm.sh/hook: pre-upgrade
    helm.sh/hook-delete-policy: "hook-succeeded,before-hook-creation"
    helm.sh/hook-weight: "1"
{{ include "vso.imagePullSecrets" .}}

Reply:
👍 updated in c1dfa0095f554f9f65e94605cbcbf0e2ea0b71a3